### PR TITLE
Add python 3.7 rebuild migrator

### PR DESCRIPTION
**Pull request by justcalamari**

### CODE REVIEW

File: conda_forge_tick/auto_tick.xsh [hunk 1]
```diff
@@ -121,6 +121,25 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
...
+    Parameters
+    ----------
+    migrators : list of Migrator
+        The list of migrators to run.
+
+    """
+
+    pygraph = copy.deepcopy(gx)
+    for node, attrs in gx.node.items():
+        if 'python' not in attrs['req']:
```
Comment (isuruf) [inline conda_forge_tick/auto_tick.xsh]: python should be in host.

File: conda_forge_tick/auto_tick.xsh [hunk 2]
```diff
@@ -121,6 +121,25 @@ def run(attrs, migrator, feedstock=None, protocol='ssh',
...
+
+    """
+
+    pygraph = copy.deepcopy(gx)
+    for node, attrs in gx.node.items():
+        if 'python' not in attrs.get('meta_yaml', {}).get('requirements', {}).get('host', []):
+            pygraph.remove_node(node)
+        if attrs.get('meta_yaml', {}).get('build', {}).get('noarch') == 'python':
+            pygraph.remove_node(node)
+    migrators.append(Rebuild(graph=pygraph, pr_limit=1, name='Python 3.7'))
```
Comment (CJ-Wright) [inline conda_forge_tick/auto_tick.xsh]: Would it be possible to handle this differently? In this upcoming round we may have to do python 3.7, compilers, and R 3.5.1. Would it be possible to build the subgraphs for each of these migrations separately and then use [compose](https://networkx.github.io/documentation/stable/reference/algorithms/generated/networkx.algorithms.operators.binary.compose.html?highlight=compose#networkx.algorithms.operators.binary.compose) (I think there is a `nx.compose` but I couldn't find it in the docs quickly). The main benefit of this is that as new migrations roll in we can `compose` the new subgraph in. I guess we can also "compose" the name of the migration by `' '.join(migration_names)`?

File: conda_forge_tick/auto_tick.xsh (hunk 1 above)
Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.xsh]: Not all the native python packages (which are not noarch) have been migrated to the new syntax so they don't have host's. Do we only care about python at build time?

**Comment by CJ-Wright** [comment]
Note that this most likely needs to try to run the compiler syntax migration before running the move to the new syntax since we won't be able to move the python projects to the new syntax otherwise.

**Pull request closed without merging**